modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -844,12 +844,31 @@ def is_series_like(self):
     prod = TreeReduce.register(pandas.DataFrame.prod)
     any = TreeReduce.register(pandas.DataFrame.any, pandas.DataFrame.any)
     all = TreeReduce.register(pandas.DataFrame.all, pandas.DataFrame.all)
-    memory_usage = TreeReduce.register(
+    # memory_usage adds an extra column for index usage, but we don't want to distribute
+    # the index memory usage calculation.
+    _memory_usage_without_index = TreeReduce.register(
         pandas.DataFrame.memory_usage,
         lambda x, *args, **kwargs: pandas.DataFrame.sum(x),
         axis=0,
     )
 
+    def memory_usage(self, **kwargs):
+        index = kwargs.get("index", True)
+        deep = kwargs.get("deep", False)
+        usage_without_index = self._memory_usage_without_index(index=False, deep=deep)
+        return (
```

Comment:
won't this error out if we do `df.memory_usage(index=True)`? since this result will be passed to base's memory usage, which will call `transpose`, and `reduce_dimension` on this, which will try to `reduce_dimension` on a dataframe with two rows - which is invalid?

Reply:
There's a passing test case with `index=True` [here](https://github.com/modin-project/modin/blob/cf9e7fcc917c7acc156c9e18747fecfc2da57cdd/modin/pandas/test/dataframe/test_reduce.py#L264-L267).  Dataframe reduce dimension will convert the result the a series, which is correct whether or not `index` is `True`.